src/app/lists/ListsPage.tsx

```diff
@@ -5,60 +5,95 @@ import { Schema } from "@/databaseUtils";
+            itemName: row.item_name,
+            ...Object.fromEntries(
+                listsHeaderKeysAndLabels
+                    .filter(([key]) => /^\d+$/.test(key))
+                    .map(([key]) => [
+                        key,
+                        {
+                            quantity: row[`quantity_for_${key}` as keyof Schema["lists"]],
+                            notes: row[`notes_for_${key}` as keyof Schema["lists"]],
+                        },
+                    ])
+            ),
+        } as ListRow; // this cast is needed here as the type system can't infer what Object.fromEntries will return
+    });
+};
+
 const ListsPage: React.FC<{}> = () => {
     const [isLoading, setIsLoading] = useState(true);
-    const [listsData, setListsData] = useState<ListRow[]>([]);
+    const [listData, setListData] = useState<ListRow[]>([]);
     const [comment, setComment] = useState("");
+    const [errorMessage, setErrorMessage] = useState<string | null>(null);
 
+    function handleSetError(error: string | null): void {
+        error ? setErrorMessage(error) : setErrorMessage(null);
+    }
     useEffect(() => {
         (async () => {
             setIsLoading(true);
-            const fetchedData = await fetchData();
-            setListsData(
-                fetchedData.data.map((row) => {
-                    const newRow = {
-                        primaryKey: row.primary_key,
-                        rowOrder: row.row_order,
-                        itemName: row.item_name,
-                        ...Object.fromEntries(
-                            listsHeaderKeysAndLabels
-                                .filter(([key]) => /^\d+$/.test(key))
-                                .map(([key]) => [
-                                    key,
-                                    {
-                                        quantity:
-                                            row[`quantity_for_${key}` as keyof Schema["lists"]],
-                                        notes: row[`notes_for_${key}` as keyof Schema["lists"]],
-                                    },
-                                ])
-                        ),
-                    } as ListRow; // this cast is needed here as the type system can't infer what Object.fromEntries will return
-                    return newRow;
-                })
-            );
-            setComment(fetchedData.comment);
+            const listData = await fetchListData();
+            setListData(formatListData(listData));
+            setComment(listData.comment);
             setIsLoading(false);
         })();
     }, []);
 
+    useEffect(() => {
+        // This requires that the DB table has Realtime turned on
+        const subscriptionChannel = supabase
+            .channel("lists-table-changes")
+            .on("postgres_changes", { event: "*", schema: "public", table: "lists" }, async () => {
+                setErrorMessage(null);
```

Comment:
I'm not sure how I feel about clearing the message every time there's a change in db. What do you think?